File: clients/android/app/src/main/kotlin/com/lextures/android/core/lms/CourseReviewLogic.kt
package com.lextures.android.core.lms

import com.lextures.android.core.navigation.MobilePlatformFeatures

object CourseReviewLogic {
    const val MAX_REVIEW_TEXT_LENGTH = 2000

    fun reviewsEnabled(features: MobilePlatformFeatures): Boolean = features.ffCourseReviews

    fun starLabel(rating: Int): String = when (rating) {
        1 -> "1 star – Poor"
        2 -> "2 stars – Fair"
        3 -> "3 stars – Average"
        4 -> "4 stars – Good"
        5 -> "5 stars – Excellent"
        else -> "Select a rating"
    }

    fun shouldShowComposer(eligibility: ReviewEligibility): Boolean =
        eligibility.eligible && (eligibility.canEdit || !eligibility.hasReview)

    fun validateRating(rating: Int): String? =
        if (rating !in 1..5) "Please select a star rating." else null

    fun validateReviewText(text: String): String? =
        if (text.length > MAX_REVIEW_TEXT_LENGTH) "Review must be $MAX_REVIEW_TEXT_LENGTH characters or fewer." else null
}

File: clients/ios/Lextures/Core/LMS/CourseReviewLogic.swift
import Foundation

/// Course review composer validation and UI state (M9.3).
enum CourseReviewLogic {
    static let maxReviewTextLength = 2000

    static func reviewsEnabled(_ features: MobilePlatformFeatures) -> Bool {
        features.ffCourseReviews
    }

    static func starLabel(rating: Int) -> String {
        switch rating {
        case 1: return L.text("mobile.reviews.star1")
        case 2: return L.text("mobile.reviews.star2")
        case 3: return L.text("mobile.reviews.star3")
        case 4: return L.text("mobile.reviews.star4")
        case 5: return L.text("mobile.reviews.star5")
        default: return L.text("mobile.reviews.selectRating")
        }
    }

    static func shouldShowComposer(_ eligibility: ReviewEligibility) -> Bool {
        eligibility.eligible && (eligibility.canEdit || !eligibility.hasReview)
    }

    static func composerTitle(hasReview: Bool, canEdit: Bool) -> String {
        if hasReview && canEdit {
            return L.text("mobile.reviews.editTitle")
        }
        return L.text("mobile.reviews.writeTitle")
    }

    static func validateRating(_ rating: Int) -> String? {
        guard (1 ... 5).contains(rating) else {
            return L.text("mobile.reviews.ratingRequired")
        }
        return nil
    }

    static func validateReviewText(_ text: String) -> String? {
        guard text.count <= maxReviewTextLength else {
            return L.format("mobile.reviews.textTooLong", maxReviewTextLength)
        }
        return nil
    }

    static func progressHint(_ percent: Int) -> String {
        L.format("mobile.reviews.progressHint", percent)
    }
}